lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2535,10 +2538,11 @@ void AttributeChecker::visitFinalAttr(FinalAttr *attr) {
 
   if (auto *accessor = dyn_cast<AccessorDecl>(D)) {
     if (!attr->isImplicit()) {
-      unsigned Kind = 2;
+      bool isProperty = false;
       if (auto *VD = dyn_cast<VarDecl>(accessor->getStorage()))
-        Kind = VD->isLet() ? 1 : 0;
-      diagnose(attr->getLocation(), diag::final_not_on_accessors, Kind)
+        isProperty = true;
+      diagnose(attr->getLocation(), diag::final_not_on_accessors,
+               isProperty)
```

Comment:
Why cannot we just use `isa<VarDecl>(accessor->getStorage())` here directly instead of `isProperty`?

Reply:
i changed it to `isProperty = isa<VarDecl>(accessor->getStorage())`

https://github.com/swiftlang/swift/pull/74462/commits/101c8074d0f1107766d7a50d7ffc4b74640ebdba

Reply:
What I meant is:

```
 diagnose(attr->getLocation(), diag::final_not_on_accessors,
                 isa<VarDecl>(accessor->getStorage()))
```

And remove `isProperty`

Reply:
done here: 
https://github.com/swiftlang/swift/pull/74462/commits/ccacc70b02b899df1b5128a7c9d6c2a802e6797b

in my day to day job - iOS Programming-  i assume that `isProperty` will help the reader understand what's happening a little bit easier, and that the compiler will optimize it anyway, this's why i preferred it initially

Reply:
No worries, the operation itself should should give the read enough context to understand what is expected here without giving it a name.